### PR TITLE
Migrate to Gradle

### PR DESCRIPTION
**Pull request by pvdstel**
**This is highly experimental. Everything worked properly after a `git clean -xdf`, which should be comparable to a freshly cloned repository.**

Some differences between Maven and Gradle:

|Maven|Gradle|
|:---|:---|
|`mvn clean`|`gradle clean`|
|`mvn test`|`gradle build` or `gradle test`|
|`mvn compile`|`gradle assemble`|

### CODE REVIEW

**Comment by mauricioaniche** [comment]
Hi @pvdstel , maybe we should make this PR green before we merge.

On the Maven build (see my travis.yml), I only run the tests for the GA module. HSQLDB tests take too much and we should figure out a way to do it in the (near) future. But for now, GA tests should run there!

**Comment by pvdstel** [comment]
You're right, we should definitely make this pull request pass Travis validation. There's currently one test failing, and I'm working on figuring out why and fixing that.

I've already updated `.travis.yml` to only let Gradle compile ga. Currently, two instructions are executed (see the diff):
- `gradle assemble`: compiles everything. This step is necessary to run the tests. It also compiles evaluation, but that isn't really a huge burden. Since there may be more subprojects in the future, I suggest everything should be compiled for now.
- `gradle :ga:test`: runs tests on the ga package. Only the tests, no other checks (like CheckStyle etc, not that we were using those)

Since the test command is specific for the ga package, the hsqldb tests aren't touched at all.

**Comment by pvdstel** [comment]
Change looks good to me!

**Comment by mauricioaniche** [comment]
@danielvangelder, give it a last check (it's good for you to see how Gradle works, as you will play a lot with it from now on). As far as I'm concerned, we can merge this one.

**Comment by pvdstel** [comment]
@mauricioaniche bad news, it's still failing on both of my machines. It used to be this (I posted this on Slack as well):
```
org.junit.ComparisonFailure: 
Expected :SELECT "COLUMN1" FROM "TABLE" WHERE "COLUMN2" >= '2017-02-09 19:02:45'
Actual   :SELECT "COLUMN1" FROM "TABLE" WHERE "COLUMN2" >= 'Thu Feb 09 15:02:45 BRST 2017'
```
and it has now become this on both my devices:
```
org.junit.ComparisonFailure: 
Expected :SELECT "COLUMN1" FROM "TABLE" WHERE "COLUMN2" >= '2017-02-09 15:02:45'
Actual   :SELECT "COLUMN1" FROM "TABLE" WHERE "COLUMN2" >= 'Thu Feb 09 15:02:45 GMT-03:00 2017'
```
I did find the cause. `SimpleDateParser` uses this pattern (`'Thu Feb 09 15:02:45 GMT-03:00 2017'`) to detect dates in the wrong format. However, this date format is also dependent on the current locale. My systems are both set to Dutch, so it expects Dutch weekdays. Awesome 😄.

Anyway, this can be fixed by changing the initializer for `wrongDatetimeFormat` to
```java
new SimpleDateFormat("EEE MMM dd HH:mm:ss z yyyy", Locale.US);
```

**Comment by mauricioaniche** [comment]
Then, the question is whether we should force the US locale on the production or in the test code. My guess would be to force it to US on the production code (as I don't expect any sane developer to generate a MySQL dump using non-US locales...), and offer users a possibility to configure it in the future.

Another thing might be to delete this feature from the system. And next time we run our experimental dataset, we fix it on the SQL queries (they don't need locale there...)

File: ga/src/main/java/nl/tudelft/serg/evosql/sql/parser/SqlSecurerVisitor.java [hunk 1]
```diff
@@ -172,18 +172,6 @@ public void visit(Parenthesis arg0) {
 
 	@Override
 	public void visit(StringValue arg0) {
```
Comment (mauricioaniche) [inline ga/src/main/java/nl/tudelft/serg/evosql/sql/parser/SqlSecurerVisitor.java]: you can remove this method too, right?

Comment (pvdstel) [reply ga/src/main/java/nl/tudelft/serg/evosql/sql/parser/SqlSecurerVisitor.java]: It's an overridden method from the `ExpressionVisitor` interface. Method itself can't be removed.

Comment (mauricioaniche) [reply ga/src/main/java/nl/tudelft/serg/evosql/sql/parser/SqlSecurerVisitor.java]: ahh, from an interface! got it!

File: build.gradle [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+        mavenLocal()
+        mavenCentral()
+        jcenter()
+    }
+
+
+    dependencies {
+        compile group: 'org.apache.logging.log4j', name: 'log4j-core', version: '2.6.2'
+        testCompile group: 'junit', name: 'junit', version: '4.12'
+        testCompile group: 'org.mockito', name: 'mockito-all', version: '2.0.2-beta'
```
Comment (TimvdLippe) [inline build.gradle]: Please use `mockito-core` with a recent stable version instead. For example https://mvnrepository.com/artifact/org.mockito/mockito-core/2.15.0 is the latest published.

**Pull request merged**